Parser/Node.cs

```diff
@@ -44,10 +54,47 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
 
+// A Read statement - to read console input
+public record NReadStmt (NIdentifier[] Identifiers) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
 // An assignment statement
 public record NAssignStmt (Token Name, NExpr Expr) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
+
+// An If statement
+public record NIfStmt (NExpr Expr, NStmt Stmt) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// An Else statement
+public record NElseStmt (NIfStmt IfStm, NStmt Stmt) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A While statement
+public record NWhileStmt (NExpr Expr, NStmt Stmt) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A Repeat statement
+public record NRepeatStmt (NStmt[] Stmts, NExpr Expr) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A Repeat statement
+public record NForStmt (NAssignStmt Assign, NExpr Expr, NStmt Stmt, bool Decrement) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}

```

Comment:
While this might be a bit clever, it will be cleaner not to reference assignment statement here. So:
`public record NForStmt (Token Var, NExpr Start, bool Increment, NExpr End, Stmt Body)`

Reply:
Changed